core/src/main/scala/org/apache/spark/sql/delta/files/TransactionalWrite.scala

```diff
@@ -295,6 +296,16 @@ trait TransactionalWrite extends DeltaLogging { self: OptimisticTransactionImpl
         statsTrackers.append(basicWriteJobStatsTracker)
       }
 
+      // Retain only known Spark writer options to avoid any potential compatibility issues
+      val options = writeOptions match {
+        case None => Map.empty[String, String]
+        case Some(writeOptions) =>
+          writeOptions.options.filterKeys(key =>
+            key.equalsIgnoreCase("maxRecordsPerFile") ||
+            key.equalsIgnoreCase(DateTimeUtils.TIMEZONE_OPTION)
```

Comment:
Adding `maxRecordsPerFile` is okay.

Do you have any use case for `timeZone`? Adding it would make the same code write different results in old and new Delta versions as old Delta versions will ignore `timeZone`. Hence, I prefer to not add it.

Could you also add a test for the change?

Reply:
Thank you for the feedback. I do not currently have a use case for `timeZone`. Removed the support for it, https://github.com/delta-io/delta/pull/1017/commits/020c16587d135b714f4ff23855d100b5696d10ca

Added a test here, https://github.com/delta-io/delta/pull/1017/commits/a47338dae1998d39fc94d621f724bff67f20c37f